index.js

```diff
@@ -220,31 +224,41 @@ var appServer = function(config) {
                     if (privateKey != undefined && certificate != undefined) {
                         var credentials = { key: privateKey, cert: certificate };
 
-                        try { //The line below can fail it the certs were generated incorrectly. But we can continue startup without HTTPS
-                            self.httpsInstance = https.createServer(credentials, self.express).listen(config.httpsPort); //create the HTTPS server
-                            self.log("Listening on HTTPS port " + config.httpsPort);
+                        try { // These two lines below can fail it the certs were generated incorrectly. But we can continue startup without HTTPS
+                            var httpsServer = https.createServer(credentials, self.express); // create the HTTPS server
+
+                            //TODO: add separate option to specify specific host address for HTTPS server to bind to ???
+                            if (typeof config.host_address === 'string') {
+                                self.httpsInstance = httpsServer.listen(config.httpsPort, config.host_address);
+                                self.log("Listening on HTTPS port " + config.httpsPort + " on address " + config.host_address);
+                            } else {
+                                self.httpsInstance = httpsServer.listen(config.httpsPort);
+                                self.log("Listening on HTTPS port " + config.httpsPort);
+                            }
                         } catch (error) {
                             self.error("Failed to listen via HTTPS Error: " + error);
                         }
                     } else {
                         self.error("Failed to load privateKey or certificate from /sslcert. HTTPS will not be enabled");
-
                     }
-
                 } else {
                     self.error("privateKey: '" + config.privateKey + "' or certificate: '" + config.certificate + "' do not exist in /sslcert. HTTPS will not be enabled");
-
                 }
             } else {
                 self.error("privatekey, httpsPort, or certificate paramater not set in config. HTTPS will not be enabled");
-
             }
-
         }
+
         // Start the server listening
         config.port = config.port || process.env.port || 80;
-        self.httpInstance = self.express.listen(config.port);
-        self.log("Listening on HTTP port " + config.port);
+
+        if (typeof config.host_address === 'string') {
+            self.httpInstance = self.express.listen(config.port, config.host_address);
+            self.log("Listening on HTTP port " + config.port + " on address " + config.host_address);
+        } else {
+            self.httpInstance = self.express.listen(config.port);
+            self.log("Listening on HTTP port " + config.port);
+        }
```

Comment:
Not really, since the [initial check](https://github.com/tejashah88/alexa-app-server/blob/master/index.js#L212) already checks if https is enabled, and if so, will then proceed to initialize `httpsInstance` [here](https://github.com/tejashah88/alexa-app-server/blob/master/index.js#L232). Otherwise, only the `httpInstance` should contain a server instance.